src/textual/layouts/horizontal.py

```diff
@@ -24,15 +25,30 @@ def arrange(
         x = max_width = max_height = 0
         parent_size = parent.size
 
-        for widget in parent.children:
-            (content_width, content_height), margin = widget.styles.get_box_model(
-                size, parent_size
-            )
-            region = Region(margin.left + x, margin.top, content_width, content_height)
-            max_height = max(max_height, content_height + margin.height)
+        box_models = [
+            widget.get_box_model(size, parent_size)
+            for widget in cast("list[Widget]", parent.children)
+        ]
+
+        margins = [
+            max((box1.margin.right, box2.margin.left))
+            for box1, box2 in zip(box_models, box_models[1:])
+        ]
+        if box_models:
+            margins.append(box_models[-1].margin.right)
+
+        x = box_models[0].margin.left if box_models else 0
```

Comment:
Afraid not, because it can still return a single placement on line 54 even with empty children. Also a single box model will produce an empty margins list on L51.